launch/docker.go

```diff
@@ -138,20 +138,72 @@ func (d *docker) setupInteractiveMode(buildEntry *buildEntry) error {
 		shellBin = "/bin/sh"
 	}
 
-	sdRunShell := fmt.Sprintf(`#!%s
-step_name="$1"
-step_dir="${SD_UTILS_DIR}/steps"
-step_list=$(ls "$step_dir")
-if [ "${step_name}" = "--list" ]; then echo "${step_list}";
-else . "${step_dir}/${step_name}"; fi
-`, shellBin)
-
-	if err := osWriteFile(fmt.Sprintf("%s/bin/sd-run", sdUtilsPath), []byte(sdRunShell), 0755); err != nil {
+	sdRunShell := strings.Join([]string{
+		fmt.Sprintf(`#!%s`, shellBin),
+		`set -o pipefail`,
+		`step_dir="${SD_UTILS_DIR}/steps"`,
+		`step_list=$(ls "$step_dir")`,
+		`if [ -z "$1" ] || [ "$1" = "--help" ]; then`,
+		`    echo "Run predefined steps of the specified job"`,
+		`    echo ""`,
+		`    echo "Usage:"`,
+		`    echo "  sdrun [flags]"`,
+		`    echo "  sdrun [step]"`,
+		`    echo ""`,
+		`    echo "Flags:"`,
+		`    echo "  --help     help for sdrun"`,
+		`    echo "  --list     show all steps"`,
+		`    echo "  --all      run all steps"`,
+		`elif [ "$1" = "--list" ]; then`,
+		`  echo "${step_list}";`,
+		`elif [ "$1" = "--all" ]; then`,
+		`  steps=$(cat "${step_dir}/.steps")`,
+		`  for step_name in $steps; do`,
+		`    cat "${step_dir}/${step_name}" | sed "s/^/${step_name}: /"`,
+		fmt.Sprintf(`    %s "${step_dir}/${step_name}" | sed "s/^/${step_name}: /"`, shellBin),
```

Comment:
Using `sh <step-file>` makes scripts closed.
So you cannot use the variable defined in step script file in the other steps.

Reply:
fixed